### ISSUE TITLE
bug: Navigation issue in sidemenu starter project

### ISSUE THREAD

**Issue by chriseen** [driftyco/ionic#4038]
**Type**: <span ionic-type>bug</span>

**Platform**: <span ionic-platform>all</span>  

<span ionic-description>I am using the latest version of Ionic (v1.0.1) to create a sidemenu starter project. Unfortunately I found an annoying issue in this "standard" template.

- Create a sidemenu starter project

    $ ionic start myApp sidemenu

- Run this project. You will see Playlists page by default. Please don't do anything... just open side menu by swiping.

- Tap on Playlists from side menu. (I know Playlists was opened already) Side menu disappears and Playlists page will appear again.

- Tap on any item, for example, Dubstep.

- You will be redirected to single playlist page. But you won't see Back button in navigation bar, as if navigation history was cleared. In this case, Back button should appear because new page (Dubstep) is pushed to navigation stack.

Navigation system and back button works well when Playlists page is opened only once via side menu. Issue happens when you access this page twice. (continuously)

I cannot actually believe this standard / well-known seed project includes such a stupid issue. And it's actually killing me.</span>

<span is-issue-template></span>

**Comment by chriseen** [comment]
It seems this issue is actual in some newer versions of Ionic, but I found this doesn't happen in v1.0.0-beta.14

**Comment by gotojmp** [comment]
I also found this bug, and I temporarily add this for bugfix:

angular.module('starter.controllers', [])
    .controller('MenuCtrl', function($scope, $ionicModal, $timeout, $rootScope, $ionicHistory) {
$rootScope.$on('$ionicSideMenuClose', function (e, side) {
            if (side == 'left') {
                $timeout(function () {
                    $ionicHistory.nextViewOptions({
                        historyRoot: false,
                        disableAnimate: false,
                        expire: null
                    });
                }, 100);
            }
        });
});

**Comment by devmaniac** [comment]
Thanks for your response, @gotojmp.  It worked.  However, I am actually planning create a pretty complicated project. I am afraid this hack will cause some unexpected troubles in my project. Is there any other official fixes or releases for this?

**Comment by stan-kondrat** [comment]
I have same problem in my production project  =(

**Comment by stan-kondrat** [comment]
If in your project *menu-close* with *href*, like in ionic-starter-sidemenu `<ion-item menu-close href="#/app/playlists">`, this fix for menuClose directive:

```javascript
IonicModule
.directive('menuClose', ['$ionicHistory', function($ionicHistory) {
  return {
    restrict: 'AC',
    link: function($scope, $element, $attr) {
      $element.bind('click', function() {
        var sideMenuCtrl = $element.inheritedData('$ionSideMenusController');
        if (sideMenuCtrl) {
          
          var currentView = $ionicHistory.currentView();
          var currentViewUrl = "#" + currentView.url;
          if($attr.href !==  currentViewUrl) {
            $ionicHistory.nextViewOptions({
              historyRoot: true,
              disableAnimate: true,
              expire: 300
            });
          }

          sideMenuCtrl.close();
        }
      });
    }
  };
}]);
```

But this check *href* of current view, will not fix  `<ion-item menu-close ng-click="login()">`

**Comment by mhartington** [comment]
Closing as a duplicate of #4132
